ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -223,6 +223,13 @@ let check_snapshot ~vmr ~op ~ref_str =
   then None
   else Some (Api_errors.vm_is_snapshot, [ref_str; Record_util.vm_operation_to_string op])
 
+let check_system_domain ~vmr ~op ~ref_str ~pbd =
+  let allowed = [`start; `start_on; `unpause; `data_source_op; `update_allowed_operations;
+                 `assert_operation_valid; `power_state_reset; `query_services; `snapshot ] in
```

Comment:
can we hide snapshot behind some development flag?

Reply:
I dropped entire special casing of driver domains, just treating them as control domains now. Which is what the documentation for control domains in the IDL says: domain 0 or driver domains.